### PR TITLE
Feature/reconcile downloaded movies

### PR DESCRIPTION
**Pull request by kdwils**
This adds an io dependency that can be used for mocking in tests

Also adding a method to the library for adding a movie file to the library by absolute path. The thought here was the download client would tell us where the download file is, and then we would do one of the following:
* if the media download is on the same file system as the library we can just rename the file to the target directory in the media library
* if the not on the same file system, we create the file in the library file system and then copy the contents over. We probably need to add a configuration here to automatically delete the download file or to leave it be.

I still need to do testing, but wanted to get this out early for any thoughts/concerns

There will be a follow up pr that will handle the download reconcile flow, but I didn't want a massive pr so I broke the library io piece out.

The flow would be the following:
* Track the download metadata from the download client so we can check the state later - completed, downloading, queued, etc.
* If complete, add the movie to the library via the manager
* Update the state of the movie to reflect that it is downloaded in the mediaz db

### CODE REVIEW

File: pkg/io/io.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+// Create copies a file from a source path to a target path
+func (o *MediaFileSystem) Copy(source, target string) (int64, error) {
+	sourceFile, err := o.Open(source)
+	if err != nil {
+		return 0, err
+	}
+	defer sourceFile.Close()
+
+	targetFile, err := o.Create(target)
```
Comment (kasuboski) [inline pkg/io/io.go:41]: What does this do if the target exists?

File: pkg/library/library_test.go [hunk 2]
```diff
@@ -124,3 +141,82 @@ func Test_fileSizeToString(t *testing.T) {
...
+		wantMovieFile := MovieFile{
+			Name: filepath.Base(tmpFile.Name()),
+			Size: 0,
+			Path: tmpFile.Name(),
+		}
+		assert.Nil(t, err)
+		assert.Equal(t, wantMovieFile, movieFile)
+	})
+
+	t.Run("different file system - error checking if same file system", func(t *testing.T) {
```
Comment (kasuboski) [inline pkg/library/library_test.go:184]: no test for successful different file system?

**Review by kasuboski — APPROVED** [review]

File: pkg/io/io.go (hunk 1 above)
Comment (kdwils) [reply pkg/io/io.go:41]: looks like the default behavior is to overwrite the contents which is maybe not the behavior we want.. I think we should probably bail out of it exists. Thoughts?

Comment (kdwils) [reply pkg/io/io.go:56]: https://github.com/kasuboski/mediaz/pull/23/commits/2d5c0cc5f26b604e7068ae47d8eae6a675b90750

**Review by kasuboski — APPROVED** [review]

**Pull request merged**